stdlib/universe/integral.go

```diff
@@ -146,37 +147,70 @@ func (t *integralTransformation) Process(id execute.DatasetID, tbl flux.Table) e
 	integrals := make([]*integral, len(cols))
 	colMap := make([]int, len(cols))
 
-	for j, c := range cols {
-		if execute.ContainsStr(t.spec.Columns, c.Label) {
-			integrals[j] = newIntegral(time.Duration(t.spec.Unit))
-
-			var err error
-			colMap[j], err = builder.AddCol(flux.ColMeta{
-				Label: c.Label,
-				Type:  flux.TFloat,
-			})
-			if err != nil {
-				return err
-			}
+	for _, c := range t.spec.Columns {
+		idx := execute.ColIdx(c, cols)
+		if idx < 0 {
+			return fmt.Errorf("column %q does not exist", c)
+		}
+
+		if tbl.Key().HasCol(c) {
+			return fmt.Errorf("cannot aggregate columns that are part of the group key")
+		}
+
+		if typ := cols[idx].Type; typ != flux.TFloat &&
+			typ != flux.TInt &&
+			typ != flux.TUInt {
+			return fmt.Errorf("cannot perform integral over %v", typ)
```

Comment:
Nice.  When I modified `derivative` I neglected to add checks like this, so I filed a bug for it.

Reply:
Apropos of nothing, it irks me that we have to do this check at run time.

Reply:
Yeah, you are right. Anyhow, I think that being schemaless would always require runtime checks on the schema.
If we had typed streams we could do any type checking at compile time and also avoid schema collisions. However we would loose flexibility. I am always open to introduce typed streams :)